### DISCUSSION TITLE
Request-Response Action

### DISCUSSION THREAD

**Discussion by NourEldin-Ali** [eclipse-glsp/glsp#1080] [Q&A] [answered]
Hi,

I added a button within the client side, and I'm trying to have a request-Response Action with the server:
1-Click on button (client side)
2-receive the action (server side)
3-send response (server side)
4-show alert that the task done (client side)

I tested with dispatch action it works but without inform the user that the task is done.
I followed [tutorial](https://eclipse.dev/glsp/documentation/actionhandler/#action-specification) but it contains some errors, and after resolve it, I'm always have an error in the server side (receive the action but directly go to exeption error).

Could you please help me, or if you can refer me to a code that already use request-response Action.

Thank you in avance for your response.

**Comment by tortmayr** [comment]
Hi @NourEldin-Ali,
what kind of error do you get? Could you please post a (partial) error log? Without that information it's kind of hard to help you?
>I followed [tutorial](https://eclipse.dev/glsp/documentation/actionhandler/#action-specification) but it contains some errors, and after resolve it, I'm always have an error in the server side (receive the action but directly go to exeption error).

What issues did you encounter? I had a quick look at the code samples and IMO they should work fine without any errors.

**Comment by rhumbertgz** [comment]
@NourEldin-Ali In case you still need it, in #1105   you can find a solution for your task  "4-show alert that the task done (client side)"

**Reply by NourEldin-Ali** [reply, marked as answer]
I was resolved the problem, by changing the KIND of the server, and change the handler of the server.
In server side:
![Screenshot 2023-11-07 113442](https://github.com/eclipse-glsp/glsp/assets/61513661/03b76b87-8ff9-4bdd-9dc1-6eef52c6c012)
![Screenshot 2023-11-07 152342](https://github.com/eclipse-glsp/glsp/assets/61513661/e162c31a-258c-49bc-93de-cc2bd2cca9fe)
![Screenshot 2023-11-07 152400](https://github.com/eclipse-glsp/glsp/assets/61513661/2d20a6b2-8ea1-435d-aaff-75f456766a9c)


In client side:
![Screenshot 2023-11-07 113737](https://github.com/eclipse-glsp/glsp/assets/61513661/76abba98-bab2-47cb-9c6f-7eab4671df5e)
![Screenshot 2023-11-07 113758](https://github.com/eclipse-glsp/glsp/assets/61513661/ba5ae4f1-2e58-402a-bd41-b8f9b73dfddc)
